src/test/optimizer/strategy/join_ordering_rule_test.cpp

```diff
@@ -0,0 +1,85 @@
+#include "gtest/gtest.h"
+
+#include "cost_model/cost_model_logical.hpp"
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "optimizer/strategy/join_ordering_rule.hpp"
+#include "statistics/column_statistics.hpp"
+#include "statistics/table_statistics.hpp"
+
+#include "strategy_base_test.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+class JoinOrderingRuleTest : public StrategyBaseTest {
+ public:
+  void SetUp() override {
+    cost_estimator = std::make_shared<CostModelLogical>();
+    rule = std::make_shared<JoinOrderingRule>(cost_estimator);
+
+    // This test only makes sure THAT something gets reordered, not what the result of this reordering is - so the stats
+    // are just dummies.
+    const auto column_statistics = std::make_shared<ColumnStatistics<int32_t>>(0.0f, 10.0f, 1, 50);
+    const auto table_statistics = std::make_shared<TableStatistics>(
+        TableType::Data, 20, std::vector<std::shared_ptr<const BaseColumnStatistics>>{column_statistics});
+
+    node_a = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}}, "a");
+    node_a->set_statistics(table_statistics);
+    node_b = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}}, "b");
+    node_b->set_statistics(table_statistics);
+    node_c = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}}, "c");
+    node_c->set_statistics(table_statistics);
+    node_d = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}}, "d");
+    node_d->set_statistics(table_statistics);
+
+    a_a = node_a->get_column("a");
+    b_a = node_b->get_column("a");
+    c_a = node_c->get_column("a");
+    d_a = node_d->get_column("a");
+  }
+
+  std::shared_ptr<MockNode> node_a, node_b, node_c, node_d;
+  LQPColumnReference a_a, b_a, c_a, d_a;
+  std::shared_ptr<AbstractCostEstimator> cost_estimator;
+  std::shared_ptr<JoinOrderingRule> rule;
+};
+
+TEST_F(JoinOrderingRuleTest, MultipleJoinGraphs) {
```

Comment:
Again, this single tests looks rather simple to me not covering many cases. Could you add some more?

Reply:
Again, for the same reasons, I don't think that's a good idea. Yes, I could add more tests and yes, that would make the code less error prone, but oh the horror when you change a bit of estimator logic. 

I'll think if the current test missed some important bit and add it, though.